core/desugarer.cpp

```diff
@@ -310,14 +317,21 @@ class Desugarer {
                     expr = alloc.make<Var>(expr->location, expr->openFodder, newSelf);
                 } else if (auto *super_index = dynamic_cast<SuperIndex*>(expr)) {
                     StringStream ss;
-                    ss << "$outer_super" << (counter++);
+                    ss << "$outer_super_index" << (counter++);
```

Comment:
Yeah String and StringStream are both char32_t.  We should probably rename them to UStringStream and UString or something.

All internal string processing is in that format, however the I/O, i.e. lexing and manifesting are in UTF-8.  So there we convert between std::string and String with explicit encode/decode functions.  Also, the Identifier object used in parameters and variables (as in the case you called out here) is also stored in UTF-8 because it is limited to [a-zA-Z_][_a-zA-Z0-9]*.  They're also interned.  Both are arbitrary decisions.

Finally, the field names as stored in the HeapObject are Identifiers (even though they can be arbitrary strings).  This technically can lead to a high-level memory leak, because being interned, they are never free'd and the set of field names is not statically bound.  This doesn't cause a probably in practice, but we probably want to fix it in the Go version.